apps/OboeTester/app/src/main/java/com/mobileer/oboetester/StreamConfigurationView.java

```diff
@@ -509,17 +512,21 @@ public void setupEffects(int sessionId) {
             mLoudnessEnhancer = new LoudnessEnhancer(sessionId);
             mLoudnessEnhancer.setTargetGain((short) mLoudnessEnhancerSeekBar.getProgress());
         } else {
-            mAcousticEchoCanceler = AcousticEchoCanceler.create(sessionId);
-            if (mAcousticEchoCanceler != null) {
-                mAcousticEchoCanceler.setEnabled(mAcousticEchoCancelerCheckBox.isEnabled());
-            } else {
-                Log.e(TAG, String.format("Could not create AcousticEchoCanceler"));
+            if (mAcousticEchoCanceler.getEnabled()) {
```

Comment:
This looks weird. We query an object to see if it is enabled.
Then we overwrite it with a new object!

Reply:
Maybe we need to keep a boolean outside the effect object that we can use before we create the object with the sessionID.

Reply:
This is set in lines 230 and 231. I agree this does looks a bit funky. I should add some comments

~~~
mAutomaticGainControlCheckBox.setEnabled(AutomaticGainControl.isAvailable());
     mAcousticEchoCancelerCheckBox.setEnabled(AcousticEchoCanceler.isAvailable());
~~~

Reply:
Still funny.

Reply:
Fixed in latest iteration